src/components/dds/textInput/TextInput.tsx

```diff
@@ -0,0 +1,153 @@
+import { componentClasses as inputClasses } from 'dds-core/components/text-input/ts/input';
+import { componentClasses as labelClasses } from 'dds-core/components/label/ts/label';
+import {
+    componentClasses as passwordClasses,
+    defaultProps
+} from 'dds-core/components/input-password/ts/input-password';
+import 'dds-core/components/text-input/_text-input.scss';
+import 'dds-core/components/label/_label.scss';
+import 'dds-core/components/input-password/_input-password.scss';
+import 'dds-core/components/form/_form.scss';
+import type { Icons, IconType } from 'dds-core/components/icon/ts/icon';
+import { useToggle } from 'hooks';
+import { Icon } from '../icon';
+
+export interface TextInputProps extends React.HTMLAttributes<HTMLDivElement> {
+    /**
+     * 	The input type to use with this component (text|url|search|tel|email)
+     *
+     */
+    type?: Type;
+    /**
+     * 	The name of a DDS icon (without the prefix *dds__*)
+     */
+    iconName?: Icons;
+    /**
+     * Define the type of icon that will be added (font-icon|svg)
+     */
+    iconType?: IconType;
+    /**
+     * Whether the text input should be disabled
+     */
+    inactive?: boolean;
+    /**
+     * Toggles the 'invalid' attribute. Used when the component is in a form with validation.
+     * @default false
+     * */
+    invalid?: boolean;
+    /**
+     * Label text for the text input
+     * */
+    labelText?: string;
+    /**
+     * Minimum length of the text input
+     * */
+    minLength?: number;
+    /**
+     * Maximum length of the text input
+     * */
+    maxLength?: number;
+    /**
+     * Name of the text input
+     * */
+    name?: string;
+    /**
+     * 	Provides a brief hint to the user as to what kind of information is expected in the field
+     * */
+    placeholder?: string;
+    /**
+     * 	Toggles the 'required' attribute. Used when the component is in a form with validation.
+     *
+     */
+    required?: boolean;
+    /**
+     * Error text for the text input
+     * */
+    errorText?: string;
+    /**
+     * Helper text for the text input
+     * */
+    helperText?: string;
+    /**
+     * Specifies the initial value for this component
+     * */
+    value?: any;
+    /**
+     * Function called on input value change
+     */
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+/**
+ * React implementation of DDS text input component
+ */
+export function TextInput({
+    className,
+    type = 'text',
+    iconName,
+    iconType,
+    inactive = false,
+    invalid,
+    labelText,
+    minLength,
+    maxLength,
+    placeholder,
+    required = true,
+    errorText,
+    helperText,
+    value,
+    onChange,
+    ...rest
+}: TextInputProps) {
+    const isPasswordType = type === 'password';
+    const [isPasswordVisible, toggleInputPassword] = useToggle(false);
+
+    return (
+        <div className={classNames(inputClasses.container, errorText ? inputClasses.isInvalid : '')} {...rest}>
```

Comment:
With `classNames` we can save few keystrokes as it handles `false` value:
```suggestion
        <div className={classNames(inputClasses.container, errorText && inputClasses.isInvalid)} {...rest}>
```

Reply:
changed: https://github.com/cloudify-cosmo/cloudify-ui-components/pull/219/commits/2426655a61312da027facd7e03b74d48a88213fa